mitodl/ocw-studio
# Umar/5124 create external resource with add links

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5124

### Description (What does it do?)
This change is aimed to add External Resources on Link Add/Update. The syntax used for Link will be replaced with syntax for Resources in Markdown i.e. `{{ % resource_link "uuid" "title" % }}`


### How can this be tested?
1. Switch to `umar/5124-create-external-resource-with-add-links` branch in `ocw-studio`
2. Open text editor
3. Add a link 
<img width="803" alt="Screenshot 2024-08-23 at 11 01 13 AM" src="https://github.com/user-attachments/assets/1c45596e-ff22-47c5-925e-5e9a574f4f49">
4. The syntax of link should be replaced with resource syntax in editor
<img width="803" alt="Screenshot 2024-08-23 at 10 58 04 AM" src="https://github.com/user-attachments/assets/113387ae-f9a0-45a2-866f-3658b41a87c2">
5. The resource should not be editable
<img width="803" alt="Screenshot 2024-08-23 at 11 01 20 AM" src="https://github.com/user-attachments/assets/1f9d9556-ccbf-4717-a957-e2ed7cbbedcc">

6. Type a URL and upon space/enter, it should convert to a Resource Link
  - If the same URL as above, verify that the existing resource ID is being used.
  - If a different URL a new resource id will be used
7. verify the use case without selecting any text and adding a link with the AddLink button


### Additional Testing
Enable/disable `OCW_STUDIO_CUSTOM_LINKUI_ENABLE` feature flag on PostHog to enable/disable the feature for users.
Setup PostHog with instruction in the Readme.
